### DISCUSSION TITLE
How to get unique listing for each NFT with GetBestListings

### DISCUSSION THREAD

**Discussion by as-iotex** [ProjectOpenSea/opensea-js#1380] [Q&A]
Hello, I want to achieve this as quick as possible:

- Get all the listed NFTs in a collection
- Get the best listing for each

The problem is that GetBestListings returns the items ordered by price. For example, let's say that:
NFT 1 is listed at 1 ETH. And it has other 99 valid listings below 2 ETH
NFT2 is listed at 2.1 ETH

Then GetBestListings() willl return the 100 listings for NFT 1 and not NFT 2

As far as I can see, there is no endpoint that returns the unique listed ids for a collection
The GetBestListings endpoint doesn't have a parameter to not return just the one best listing for each unique NFT

I am afraid this could be a limitation of the API rather that the JS library. Am I missing something?

**Comment by ryanio** [comment]
correct there is no current endpoint that returns unique listed ids for a collection. you'll want to paginate get best listings. there should usually be not too many duplicate listings

**Reply by as-iotex** [reply]
Thanks very much. I will waitl until you add pagination to GetBestOrders
